### PR TITLE
Form Objects based on prototype version 1

### PR DESCRIPTION
**Pull request by autotelik**
Create form objects for individual journey based on prototype version 1

Create one form object per state, and initialize with best guess at model and attributes, ready to be refined once the modelling redone

### CODE REVIEW

**Comment by autotelik** [comment]
:eyes:

**Comment by autotelik** [comment]
:eyes:   ping ... this stuff needed asap so we can continue dev on the forms/views

File: app/forms/flood_risk_engine/steps/applicant_contact_form.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module FloodRiskEngine
+  module Steps
+    class ApplicantContactForm < BaseForm
+
```
Comment (woodpigeon) [inline app/forms/flood_risk_engine/steps/applicant_contact_form.rb]: Whats the step with this one Tom?

File: app/forms/flood_risk_engine/steps/applicant_contact_telephone_form.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  module Steps
+    class ApplicantContactTelephoneForm < BaseForm
+
+      # Define the attributes on the inbound model, that you want included in your form/validation with
+      # property :name
+      # For full API see  - https://github.com/apotonick/reform
+        property :phone_number
+
+      def self.factory(enrollment)
+        contact = enrollment.correspondence_contact || ::Contact.new
```
Comment (woodpigeon) [inline app/forms/flood_risk_engine/steps/applicant_contact_telephone_form.rb]: should that be applicant contact?

**Comment by autotelik** [comment]
@woodpigeon  Just to confirm, each step is named after associated page.url in prototype so yes state/step = :applicant_contact 

FYI - form class is currently calculated from state.classify ... ie "FloodRiskEngine::Steps::#{@enrollment.state.classify}Form".constantize.factory(@enrollment)

But that's just to get my local state engine controller working so can change as you see fit

**Pull request merged**